agent/internal/fluent_intg_test.go

```diff
@@ -0,0 +1,257 @@
+//+build integration
+
+package internal
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/google/uuid"
+	"github.com/labstack/echo"
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+	aproto "github.com/determined-ai/determined/master/pkg/agent"
+	"github.com/determined-ai/determined/master/pkg/archive"
+	cproto "github.com/determined-ai/determined/master/pkg/container"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/test/testutils"
+)
+
+func TestFluentPostgresLogging(t *testing.T) {
+	// GIVEN a mock master that accepts logs
+	logBuffer, cl := mockLogAcceptor(t)
+	defer cl()
+
+	// AND a successfully started fluent actor
+	aConf := defaultAgentConfig()
+	f, err := newFluentActor(aConf, aproto.MasterSetAgentOptions{
+		MasterInfo: aproto.MasterInfo{},
+		LoggingOptions: model.LoggingConfig{
+			DefaultLoggingConfig: &model.DefaultLoggingConfig{},
+		},
+	})
+	assert.NilError(t, err, "error starting fluentbit")
+	sys := actor.NewSystem("")
+	sys.MustActorOf(actor.Addr("fluent"), f)
+
+	// AND a container that prints some predefined logs
+	trialID := math.MaxInt16 + rand.Int31n(math.MaxInt16)
+	expected, actual := makeLogTestCase(int(trialID), aConf.AgentID)
+	runContainerWithLogs(t, actual, int(trialID), f.port)
+
+	// THEN fluent should parse all fields as expected and ship them to the mock master.
```

Comment:
non-blocking: I haven't seen this style of test comments before, but it seems like there should be a WHEN somewhere here?

Reply:
yep, i had a typo. yeah, it's sort of common https://martinfowler.com/bliki/GivenWhenThen.html it just helps me be more clear about the intent of the tests.